app/models/supplejack_api/search.rb

```diff
@@ -66,18 +66,27 @@ def execute_solr_search
 
     # The records that match the criteria within each role will be removed
     # from the search results
-    #
-    def self.role_collection_restrictions(scope)
+    def self.role_collection_exclusions(scope)
+      role = scope&.role&.to_sym
+      return [] if scope.blank? || schema_class.roles[role].record_exclusions.blank?
```

Comment:
This code is causing code quality issues. Need to refactor, let me know if you have suggestions :)